api/src/main/java/org/openmrs/module/fhir2/providers/r3/DiagnosticReportFhirResourceProvider.java

```diff
@@ -76,6 +77,19 @@ public MethodOutcome updateDiagnosticReport(@IdParam IdType id, @ResourceParam D
 		    diagnosticReportService.update(idPart, DiagnosticReport30_40.convertDiagnosticReport(diagnosticReport)));
 	}
 	
+	@Delete
+	@SuppressWarnings("unused")
+	public OperationOutcome deleteDiagnosticReport(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.DiagnosticReport diagnosticReport = diagnosticReportService.delete(id.getIdPart());
+		if (diagnosticReport == null) {
+			throw new ResourceNotFoundException("Could not find medication to delete with id " + id.getIdPart());
+		}
+		OperationOutcome retVal = new OperationOutcome();
+		retVal.setId(id.getIdPart());
+		retVal.getText().setDivAsString("Deleted successfully");
+		return retVal;
```

Comment:
Instead of the above code, leverage the util method provided by `FhirProviderUtils.buildDelete()` introduced #214 
```java
return FhirProviderUtils.buildDelete(diagnosticReport);
```

Reply:
sure I will do